users/script.js

```diff
@@ -474,18 +485,92 @@ function getCheckedValues(groupName) {
 
 function getFilteredUsersURL(checkedValuesSkills, checkedValuesAvailability) {
   const params = new URLSearchParams();
-
   checkedValuesSkills.forEach((skill) => {
     params.append('tagId', skill);
   });
-
   checkedValuesAvailability.forEach((availability) => {
     params.append('state', availability);
   });
-
   return `?${params.toString()}`;
 }
 
+function manipulateQueryParamsToURL(constructedQueryParam) {
+  const currentURLInstance = new URL(window.location.href);
+  currentURLInstance.search = '';
+  const currentURL = currentURLInstance.href;
+  if (constructedQueryParam) {
+    const newURLWithQueryParams = `${currentURL}${constructedQueryParam}`;
+    window.history.pushState(
+      { path: newURLWithQueryParams },
+      '',
+      newURLWithQueryParams,
+    );
+  } else {
+    const newUrlWithoutQueryParams = `${currentURL}`;
+    window.history.pushState(
+      { path: newUrlWithoutQueryParams },
+      '',
+      newUrlWithoutQueryParams,
+    );
+  }
+}
+
+function parseQueryParams() {
+  const urlString = window.location.href;
+  const urlObjInstance = new URL(urlString);
+  const queryParamsObj = urlObjInstance.searchParams.entries();
+  const queryObject = {};
+  for (const [key, value] of queryParamsObj) {
+    if (!queryObject[key]) {
+      queryObject[key] = [];
+    }
+    if (key === 'state') {
+      queryObject[key].push(value.toLocaleUpperCase());
+    } else {
+      queryObject[key].push(value);
+    }
+  }
+  return queryObject;
+}
+
+function selectFiltersBasedOnQueryParams(checkboxes) {
+  const parsedQuery = parseQueryParams();
+  const checkBoxArray = Array.from(checkboxes);
+  checkBoxArray.map((checkbox) => {
+    for (let key in parsedQuery) {
+      const checkedValues = parsedQuery[key];
+      checkedValues.map((value) => {
+        if (checkbox.id === value) {
+          checkbox.checked = true;
+        }
+      });
+    }
+  });
```

Comment:
- Array.from give us a shallow copy of the original array, changing the nested values might cause a change to the original array can we cross-check this one? 
- Can we think of any other way to set the `checked` status of the filters for each query param?
- If you just want to iterate over an array I think using a `forEach` loop would be more accurate, also for iterating over an object where you only need the keys for it you can make use of Object.entries.

- Please add a JSDoc comment stating the type of `checkboxes` param the function is taking.

Reply:
- Optimized this function in the recent commit.
### JsDoc comment stating the type of checkbox(changed to filters) :
`@param` {NodeList} filters - A NodeList of filter elements with 'id' and 'checked' properties.